libs/sdk-ui-vis-commons/src/legend/StaticLegend.tsx

```diff
@@ -134,10 +149,30 @@ export class StaticLegend extends React.PureComponent<IStaticLegendProps> {
                         onItemClick={onItemClick}
                     />
                 </div>
-                {hasPaging && this.renderPaging(visibleItemsCount)}
+                {hasPaging && this.renderPaging(pagesCount)}
             </div>
         );
     }
+
+    private getPagesCount(seriesLength: number, visibleItemsCount: number) {
+        const defaultPagesCount = Math.ceil(seriesLength / visibleItemsCount);
+        return this.hasCustomComponent() ? defaultPagesCount + 1 : defaultPagesCount;
+    }
+
+    private getStart(page: number, visibleItemsCount: number) {
```

Comment:
rename to getPagedSeriesStart/End to explicitly show, that there can be custom page on the beginning?

Reply:
Done https://github.com/gooddata/gooddata-ui-sdk/pull/1341/commits/675e10976b492226752cddb9482104b68f184c73